packages/toolpad-app/src/toolpad/propertyControls/SelectOptions.tsx

```diff
@@ -95,19 +109,30 @@ function SelectOptionsPropEditor({
         variant="outlined"
         color="inherit"
         fullWidth
-        onClick={() => setEditOptionsDialogOpen(true)}
+        onClick={() => {
+          setEditOptionsDialogOpen(true);
+        }}
       >
         {label}
       </Button>
       <Dialog
         fullWidth
         open={editOptionsDialogOpen}
-        onClose={() => setEditOptionsDialogOpen(false)}
+        onClose={() => {
+          if (optionTextInputInvalid) {
```

Comment:
Are we blocking closing the dialog? Maybe could be frustrating for the user, we could maybe make some assumption instead and close the dialog.

Reply:
Makes sense. No need to add friction unnecessarily - present behaviour is to just have an undefined value in there and that should be fine